File: Advanced_Data_Types/SinglyLinkedLists/linked_list.py


class Node:
    def __init__(self, data):
        self.data = data
        self.next = None




class LinkedList:
    def __init__(self):
        self.head = None

    
    def add_front(self, value):
        new_node = Node(value)
        if self.head is None:
            self.head = new_node
            return self
        new_node.next = self.head
        self.head = new_node
        return self
    
    
    def remove_front(self):
        if self.head is None:
            print("Linked List is empty")
            return self
        removed_data = self.head.data
        self.head = self.head.next
        return removed_data
    
    def print_head_data(self):
        return self.head.data
    
    def find_sum(self):
        runner = self.head
        sum = 0
        while runner != None:
            sum = sum + runner.data
            runner = runner.next
        return sum
    
    def contains_value(self, num):
        runner = self.head
        while runner != None:
            if runner.data == num:
                return True
            runner = runner.next
        return False
    
    def node_count(self):
        runner = self.head
        count = 0
        while runner != None:
            count += 1
            runner = runner.next
        return count



    
linked_list = LinkedList()
print(linked_list.add_front(15))
print(linked_list.add_front(16))
print(linked_list.add_front(17))
print(linked_list.add_front(18))
print(linked_list.add_front(18))
print(linked_list.find_sum())
print(linked_list.contains_value(14))
print(linked_list.node_count())

File: Advanced_Data_Types/SinglyLinkedLists/linkedList.js



//! Singly Linked Lists 
class Node {
    constructor(data) {
        this.data = data
        this.next = null
    }
}

class LinkedList {
    constructor() {
        this.head = null
    }

    addFront(val) {
        let new_node = new Node(val)
        if (!this.head) {
            this.head = new_node
            return this
        }
        new_node.next = this.head
        this.head = new_node
        return this
    }

    removeFront() {
        // Checking to make sure the list is not empty
        if (this.head === null) {
            console.log("Linked list is empty. Nothing to remove.")
            return this
        }
        // Storing data that is going to be removed in a variable
        const removedData = this.head.data
        // Adjusting the head of the linked list
        this.head = this.head.next
        // Returning the removed data
        return removedData
    }

    getHeadData() {
        return this.head.data
    }

    getSumOfList() {
        let sum = 0
        let runner = this.head
        while(runner !== null) {
            sum += runner.data
            runner = runner.next
        }
        return sum
    }

    containsValue(num) {
        let runner = this.head
        while (runner !== null) {
            if (runner.data == num) {
                return true
            }
            runner = runner.next
        }
        return false
    }

    nodeCount() {
        let runner = this.head
        let count = 0
        while (runner !== null) {
            count++
            runner = runner.next
        }
        return count
    }
}

myLinkedList = new LinkedList()

myLinkedList.addFront(17)
myLinkedList.addFront(16)
myLinkedList.addFront(15)
myLinkedList.addFront(14)
console.log(myLinkedList.addFront(14))
console.log("Head Data: ", myLinkedList.getHeadData())


